iam-policy-autopilot-mcp-server/src/mcp.rs

```diff
@@ -166,14 +171,52 @@ impl ServerHandler for IamAutoPilotMcpServer {
                 ..Default::default()
         }
     }
+
+    async fn list_tools(
+        &self,
+        _: Option<PaginatedRequestParam>,
+        _: RequestContext<RoleServer>,
+    ) -> Result<ListToolsResult, McpError> {
+        let all_tools = self.tool_router.list_all();
+
+        // Filter out fix_access_denied tool when in read-only mode
+        let tools: Vec<Tool> = if self.read_only {
+            all_tools
+                .into_iter()
+                .filter(|tool| tool.name.as_ref() != "fix_access_denied")
```

Comment:
Instead of filtering by name, would it make sense for the tools to have a property against we filter?